### PR TITLE
Update supported puppet versions

### PR DESCRIPTION
**Pull request by tbrisker**
Also remove deprecated legacy_puppet_hostname setting and correct wrong
server_use_legacy_use_con setting

### CODE REVIEW

File: _includes/manuals/1.19/3.1.3_puppet_versions.md [hunk 1]
```diff
@@ -73,7 +66,7 @@ Puppet Server is the application for serving Puppet agents used by default in Pu
   <tr>
     <td>2.0 - 2.1</td>
```
Comment (tbrisker) [inline _includes/manuals/1.19/3.1.3_puppet_versions.md]: @ekohl - would love your input regarding the puppetserver versions we should be supporting.

File: _includes/manuals/1.12/3.1.3_puppet_versions.md [hunk 2]
```diff
@@ -86,7 +86,7 @@ Puppet Server is the application for serving Puppet agents used by default in Pu
   <tr>
     <td>2.0 - 2.1</td>
     <td>Limited support</td>
-    <td>Requires server_use_legacy_use_conf, server_puppetserver_version to be set</td>
+    <td>Requires server_use_legacy_host_conf, server_puppetserver_version to be set</td>
```
Comment (ekohl) [inline _includes/manuals/1.12/3.1.3_puppet_versions.md]: It's actually `server_use_legacy_auth_conf`

File: _includes/manuals/1.19/1.2_release_notes.md [hunk 3]
```diff
@@ -292,6 +292,7 @@ Previously, the cloud-init integration for oVirt supported only a sub-set of clo
 ### Deprecations
 * Two-pane layout has been deprecated and will be removed in 1.20.
 * The `notice` function for flashing messages has been deprecated in favor of `success` or `info` and will be removed in 1.20.
+* The deprecated `legacy_puppet_hostname` setting has been removed.
```
Comment (ekohl) [inline _includes/manuals/1.19/1.2_release_notes.md]: This isn't really a deprecation. I'd put it under Upgrade warnings instead.

File: _includes/manuals/1.19/3.1.3_puppet_versions.md [hunk 4]
```diff
@@ -10,19 +10,12 @@ Foreman 1.18 is the last version that will support Puppet < 4.0. Report/fact pro
     <th><a href="/manuals/{{page.version}}/index.html#3.5.5FactsandtheENC">External node classifier</a></th>
   </tr>
   <tr>
-    <td>0.x-3.7.x</td>
+    <td>0.x-3.x</td>
+    <td>Not supported</td>
     <td>Not supported</td>
-    <td>Untested</td>
```
Comment (ekohl) [inline _includes/manuals/1.19/3.1.3_puppet_versions.md]: Technically it'll probably still work as no code has been removed so I'd keep this on Untested.

File: _includes/manuals/1.19/3.1.3_puppet_versions.md (hunk 1 above)
Comment (ekohl) [reply _includes/manuals/1.19/3.1.3_puppet_versions.md]: Right now no code has been removed so it'll probably still work.

File: _includes/manuals/1.19/3.1.3_puppet_versions.md [hunk 5]
```diff
@@ -73,7 +66,7 @@ Puppet Server is the application for serving Puppet agents used by default in Pu
   <tr>
     <td>2.0 - 2.1</td>
     <td>Limited support</td>
-    <td>Requires server_use_legacy_use_conf, server_puppetserver_version to be set</td>
+    <td>Requires server_use_legacy_host_conf, server_puppetserver_version to be set</td>
   </tr>
   <tr>
     <td>2.2 - 2.7</td>
```
Comment (ekohl) [inline _includes/manuals/1.19/3.1.3_puppet_versions.md]: 2.8 is out and should also work. Maybe we should mark < 2.8 as Untested and 2.8 as Supported. @mmoll?

Comment (mmoll) [reply _includes/manuals/1.19/3.1.3_puppet_versions.md]: sounds legit :+1:

File: _includes/manuals/1.12/3.1.3_puppet_versions.md (hunk 2 above)
Comment (tbrisker) [reply _includes/manuals/1.12/3.1.3_puppet_versions.md]: Do these settings actually exist? I can't find them anywhere, not even when digging in ancient version of foreman.

Comment (tbrisker) [reply _includes/manuals/1.12/3.1.3_puppet_versions.md]: oh nevermind, i realized this is actually a setting on the puppetserver.

**Comment by tbrisker** [comment]
@ekohl updated according to comments

**Review by ekohl — APPROVED** [review]

**Pull request merged**